lib/healthcheck/worker.go

```diff
@@ -67,6 +68,16 @@ func (cfg *workerConfig) checkAndSetDefaults() error {
 
 // newWorker creates and starts a new worker.
 func newWorker(ctx context.Context, cfg workerConfig) (*worker, error) {
+	w, err := newUnstartedWorker(ctx, cfg)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	go w.run(ctx)
+	return w, nil
+}
+
+// newUnstartedWorker creates a worker without running it.
+func newUnstartedWorker(ctx context.Context, cfg workerConfig) (*worker, error) {
```

Comment:
`newWorker` will be called from `AddTarget(ctx context.Context, target Target) error`, but given that `manager` now contains `closeContext` I wonder if we could or should simplify the signature to ``AddTarget(target Target) error` with removed context using the `closeContext` from `manager`? 

I'm not sure how the usage pattern will look like to tell if this is a good or bad idea, but I wanted to bring this to your consideration.